server/Arcadia.Assistant.Notifications.Push/PushNotificationsActor.cs

```diff
@@ -0,0 +1,141 @@
+﻿namespace Arcadia.Assistant.Notifications.Push
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Threading.Tasks;
+
+    using Akka.Actor;
+    using Akka.Event;
+
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Serialization;
+
+    using Arcadia.Assistant.Configuration.Configuration;
+
+    public class PushNotificationsActor : UntypedActor, ILogReceive
+    {
+        private readonly IPushSettings pushSettings;
+        private readonly IHttpClientFactory httpClientFactory;
+
+        public readonly ILoggingAdapter logger = Context.GetLogger();
+
+        public PushNotificationsActor(IPushSettings pushSettings, IHttpClientFactory httpClientFactory)
+        {
+            this.pushSettings = pushSettings;
+            this.httpClientFactory = httpClientFactory;
+        }
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case PushNotification msg when this.pushSettings.Enabled:
+                    this.SendPushNotification(msg)
+                        .PipeTo(
+                            this.Self,
+                            success: x => new SendPushNotificationResult(x),
+                            failure: err => new SendPushNotificationResultFinish(err.Message));
+                    break;
+
+                case SendPushNotificationResult msg:
+                    this.HandleSendPushNotificationResults(msg.Responses)
+                        .PipeTo(
+                            this.Self,
+                            success: () => SendPushNotificationResultFinish.Instance,
+                            failure: err => new SendPushNotificationResultFinish(err.Message));
+                    break;
+
+                case SendPushNotificationResultFinish msg:
+                    if (msg.ErrorMessage != null)
+                    {
+                        this.logger.Warning(msg.ErrorMessage);
+                    }
+
+                    break;
+            }
+        }
+
+        private Task<HttpResponseMessage[]> SendPushNotification(PushNotification message)
+        {
+            this.logger.Debug("Push notification message received");
+
+            var jsonMessage = this.SerializeNotification(message);
+            this.logger.Debug($"Serialized notification message: {jsonMessage}");
+
+            var requests = this.pushSettings.ApplicationPushUrls
+                .Select(url => this.SendPushNotificationRequest(url, jsonMessage));
+
+            return Task.WhenAll(requests);
+        }
+
+        private async Task HandleSendPushNotificationResults(IEnumerable<HttpResponseMessage> responses)
+        {
+            foreach (var response in responses)
+            {
+                if (response.StatusCode == HttpStatusCode.Accepted)
```

Comment:
don't check for accepted, check for 2xx status code

Reply:
Done.